backend/curriculum_tracking/tests/test_api_views.py

```diff
@@ -234,6 +239,45 @@ def test_set_project_link(self):
         project.refresh_from_db()
         self.assertEqual(project.link_submission, link_2)
 
+    @mock.patch.object(RecruitProject, "setup_repository")
+    def test_setup_project_repo_call_from_api_action_option(self, setup_repository):
+
+        JAVASCRIPT = "js"
+        super_user = factories.UserFactory(is_superuser=True)
+
+        def get_repo_mock(
+            github_auth_login="", repo_full_name="", api=None, response404=None, add_collaborators=None
+        ):
+            return {
+                "full_name": f"me/{repo_full_name}",
+                "owner": {"login": "me"},
+                "ssh_url": f"https://{repo_full_name}.git",
+                "private": True,
+                "created_at": timezone.now().strftime(GITHUB_DATETIME_FORMAT),
+                "archived": False,
+            }
+
+        content_item = factories.ContentItemFactory(
+            content_type=ContentItem.PROJECT,
+            project_submission_type=ContentItem.REPOSITORY,
+            flavours=[JAVASCRIPT],
+        )
+
+        card = factories.AgileCardFactory(
+            content_item=content_item,
+            flavours=[JAVASCRIPT],
+            status=AgileCard.READY,
+        )
+
+        assignee = SocialProfileFactory().user
+        card.assignees.set([assignee])
+        project = card.recruit_project
+        url = f"{self.get_list_url()}{card.id}/setup_project_repo/"
+        self.login(super_user)
+        response = self.client.post(path=url, data={"card_id": card.id})
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(project.setup_repository.assert_called)
```

Comment:
project.setup_repository.assert_called is a function. Asserting that it is True is meaningless because functions are truthy.

```
def foo():pass
self.assertTrue(foo) # this never fails
```

If we use the method correctly then this is what happens:

```
project.setup_repository.assert_called()
*** AssertionError: Expected 'setup_repository' to have been called.
```

So the tests think it wasn't called. 

It was called in a seperate worker process. That seperate worker process doesn't know it is unit testing anything, it's just a worker. So it doesn't know that you are trying to mock anything. So it doesn't even record what got called and what didn't.